mapper/dm_agent/src/error.rs

```diff
@@ -0,0 +1,32 @@
+use crate::collectd::CollectdError;
+use mqtt_client::Error as MqttClientError;
+use std::sync::Arc;
+use thin_edge_json::{
+    group::{MeasurementGrouper, MeasurementGrouperError},
+    serialize::ThinEdgeJsonSerializationError,
+};
+use tokio::sync::mpsc::error::SendError;
+
+#[derive(thiserror::Error, Debug)]
+pub enum DeviceMonitorError {
+    #[error(transparent)]
+    MqttClientError(#[from] Arc<MqttClientError>),
```

Comment:
That's because the `ErrorStream` in our MQTT library throws an `Arc<Error>` on calling `error_stream.next()`. Unfortunately, the wrapped error in that `Arc` is a reference and not an owned one either because of which I couldn't pass that reference to this error enum variant. Otherwise I'd have had to put additional life time annotation on this whole error type, which would have overcomplicated the whole code for little benefit.

Reply:
Removing the `Arc` and the `impl From<mqtt_client::Error> for DeviceMonitorError` is working fine without any lifetime annotation.